docs/sphinx/applications/python/adapt-vqe.ipynb

```diff
@@ -0,0 +1,449 @@
+            "        count+=1\n", "\n",
+            "    for i in range(0, len(coef_double), 8):\n",
+            "        exp_pauli(coef_double[i] * theta[count], q, pool_double[i])\n",
+            "        exp_pauli(coef_double[i+1] * theta[count], q, pool_double[i+1])\n",
+            "        exp_pauli(coef_double[i+2] * theta[count], q, pool_double[i+2])\n",
+            "        exp_pauli(coef_double[i+3] * theta[count], q, pool_double[i+3])\n",
+            "        exp_pauli(coef_double[i+4] * theta[count], q, pool_double[i+4])\n",
+            "        exp_pauli(coef_double[i+5] * theta[count], q, pool_double[i+5])\n",
+            "        exp_pauli(coef_double[i+6] * theta[count], q, pool_double[i+6])\n",
+            "        exp_pauli(coef_double[i+7] * theta[count], q, pool_double[i+7])\n",
+            "        count+=1"
+        ]
+    }, {
+        "cell_type": "markdown",
+        "metadata": {},
+        "source": ["## Beginning of ADAPT-VQE:"]
+    }, {
+        "cell_type":
+            "code",
+        "execution_count":
+            16,
+        "metadata": {},
+        "outputs": [{
+            "name":
+                "stdout",
+            "output_type":
+                "stream",
+            "text": [
+                "Beginning of ADAPT-VQE\n", "Step:  0\n",
+                "Norm of the gradient:  0.36242092403039405\n",
+                "max_grad:  0.36242092403039405\n",
+                "Selected pool at current step:  [['XXYX', 'XXXY', 'XYYY', 'YYXY', 'XYXX', 'YXYY', 'YYYX', 'YXXX']]\n",
+                "pool single:  []\n", "coef_single:  []\n",
+                "pool_double:  ['XXYX', 'XXXY', 'XYYY', 'YYXY', 'XYXX', 'YXYY', 'YYYX', 'YXXX']\n",
+                "coef_double:  [0.125, 0.125, 0.125, -0.125, -0.125, 0.125, -0.125, -0.125]\n",
+                "tot_single:  0\n", "tot_double:  1\n", "theta_single []\n",
+                "theta_double:  [0.0]\n", "theta [0.0]\n",
+                "Optmized Energy:  -1.1372838344885003\n",
+                "Optimizer exited successfully:  True\n",
+                "CONVERGENCE: REL_REDUCTION_OF_F_<=_FACTR*EPSMCH\n",
+                "dE:  -1.1372838344885003\n", "\n", "\n", "Step:  1\n",
+                "Norm of the gradient:  1.1233558207257577e-07\n", "\n",
+                " Final Result:  \n", "\n",
+                "Final parameters:  [0.11278279920936399]\n",
+                "Selected pools:  [['XXYX', 'XXXY', 'XYYY', 'YYXY', 'XYXX', 'YXYY', 'YYYX', 'YXXX']]\n",
+                "Number of pools:  1\n", "Final energy:  -1.1372838344885003\n"
+            ]
+        }],
+        "source": [
+            "from scipy.optimize import minimize\n", "\n",
+            "print('Beginning of ADAPT-VQE')\n", "\n", "threshold=1e-3\n",
+            "E_prev=0.0\n", "e_stop=1e-5\n", "init_theta=0.0\n", "\n",
+            "theta_single=[]\n", "theta_double=[]\n", "\n", "pool_single=[]\n",
```

Comment:
I think our coding convention has spaces before and after '='. (at least that's what I see in previous code)
'=' with no spaces (or space on just one side) appears in other parts of the code - should be changed.